src/bson.rs

```diff
@@ -53,25 +53,20 @@ pub enum Bson {
     Boolean(bool),
     /// Null value
     Null,
-    /// Regular expression - The first cstring is the regex pattern, the second is the regex
-    /// options string. Options are identified by characters, which must be stored in
-    /// alphabetical order. Valid options are 'i' for case insensitive matching, 'm' for
-    /// multiline matching, 'x' for verbose mode, 'l' to make \w, \W, etc. locale dependent,
-    /// 's' for dotall mode ('.' matches everything), and 'u' to make \w, \W, etc. match
-    /// unicode.
-    RegExp(String, String),
+    /// Regular expression
+    RegExp(RegExp),
     /// JavaScript code
     JavaScriptCode(String),
     /// JavaScript code w/ scope
-    JavaScriptCodeWithScope(String, Document),
+    JavaScriptCodeWithScope(JavaScriptCodeWithScope),
     /// 32-bit integer
     I32(i32),
     /// 64-bit integer
     I64(i64),
     /// Timestamp
     TimeStamp(i64),
```

Comment:
I think it would be far less error-prone and more ergonomic to introduce a type for timestamp as well rather than bit shifting an integer.

Reply:
Yeah, I think this is preferable too. The original implementation probably relied a little too heavily on mirroring the binary structure exactly. I'll add a follow-up commit for this.